wire-ios/Wire-iOS/Sources/UserInterface/Calling/CallController/ActiveCallRouter.swift

```diff
@@ -151,22 +165,56 @@ extension ActiveCallRouter: ActiveCallRouterProtocol {
     }
 
     // MARK: - Alerts
-    func presentSecurityDegradedAlert(for reason: CallDegradationReason) {
-        executeOrSchedulePostCallAction { [weak self] in
+
+    func presentSecurityDegradedAlert(for reason: CallDegradationReason, callEnded: Bool = false, completion: @escaping (AlertChoice) -> Void) {
```

Comment:
Maybe we could split this into separate methods for different types of degradation alerts, e.g `presentAlertForIncomingCallInDegradedConversation` and `presentAlertForDegradedCall`, what do you think?